provider/k8s/k8s.go

```diff
@@ -258,9 +273,32 @@ func (p *Provider) initializeTemplates() error {
 		return errors.WithStack(err)
 	}
 
+	d, _ := p.Cluster.AppsV1().Deployments("cert-manager").Get("cert-manager", am.GetOptions{})
+
+	// println(d.Spec.Template.Labels["app.kubernetes.io/version"])
+
 	if p.CertManager {
-		if err := p.applySystemTemplate("cert-manager", nil); err != nil {
-			return errors.WithStack(err)
+		if d != nil && d.Spec.Template.Labels["app.kubernetes.io/version"] != "v1.9.0" {
+			println("updating cert-manager...")
+			p.deleteSystemTemplate("cert-manager", nil)
+
+			// println(">>> waiting for namespace to be deleted")
+			// TODO Query the cert-manager namespace and return when it's deleted
+			time.Sleep(time.Second * 20)
```

Comment:
we should poll for the namespace and proceed when it's not found, i.e deleted.
`n, err := p.Cluster.CoreV1().Namespaces().Get("cert-manager", am.GetOptions{})`

`n == nil` means the namespace was not found and we can proceed.